rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -45,11 +46,13 @@ pub trait MetadataBuilder {
 
 #[derive(Clone, new)]
 pub struct BaseMetadataBuilder {
-    chain_setup: ChainConf,
-    prover_sync: Arc<RwLock<MerkleTreeBuilder>>,
-    validator_announce: Arc<dyn ValidatorAnnounce>,
-    allow_local_checkpoint_syncers: bool,
-    metrics: Arc<CoreMetrics>,
+    pub chain_setup: ChainConf,
+    pub prover_sync: Arc<RwLock<MerkleTreeBuilder>>,
+    pub validator_announce: Arc<dyn ValidatorAnnounce>,
+    pub allow_local_checkpoint_syncers: bool,
+    pub metrics: Arc<CoreMetrics>,
+    pub depth: u32,
+    pub max_depth: u32,
```

Comment:
We don't we need to derive "new"? We use it here: `agents/relayer/src/relayer.rs:158`

Reply:
I've exposed functions instead of properties

Reply:
If the fields are public, there is no need for a constructor because it becomes better to just do `BaseMetadataBuillder { .. }` since then the fields are named and not ordered.